### ISSUE TITLE
[PRE REVIEW]: Bruker2nifti: Magnetic Resonance Images converter from Bruker ParaVision to Nifti format

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#342]
**Submitting author:** @SebastianoF (<a href="http://orcid.org/0000-0001-9279-4765">Sebastiano Ferraris</a>)
**Repository:** <a href="https://github.com/SebastianoF/bruker2nifti" target ="_blank">https://github.com/SebastianoF/bruker2nifti</a>
**Version:** 1.0.0
**Editor:** @Kevin-Mattheus-Moerman
**Reviewer:** @rougier

**Author instructions**

Thanks for submitting your paper to JOSS @SebastianoF. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@SebastianoF if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://github.com/openjournals/joss/blob/master/docs/reviewers.csv) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by SebastianoF** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the reviewer of this submission
@whedon assign @username as reviewer

# List the GitHub usernames of the JOSS editors
@whedon list editors

# List of JOSS reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Open the review issue
@whedon start review
```

:construction: **Important** :construction:

This is all quite new. Please make sure you check the top of the issue after running a @whedon command (you might also need to refresh the page to see the issue update).

**Comment by SebastianoF** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of JOSS reviewers: https://github.com/openjournals/joss/blob/master/docs/reviewers.csv

**Comment by SebastianoF** [comment]
Hello!
@arokem @rougier @sealhuang @oesteban @effigies @nirum @ahwillia - you are all in neuroimage/neuroscience + Python - Is there anyone interested in peer-reviewing bruker2nifti as submitted for JOSS? 
It is not fundamental, but it may help if you have worked with Bruker ParaVision software/formats before.

@neurolabusc @matthew-brett do you know someone potentially interested in peer-reviewing the code bruker2nifti?

Thanks!

**Comment by effigies** [comment]
I don't have any experience with Bruker, so if any other potential reviewers do, you should probably prioritize them over me. That said, as long as there are some good and bad Bruker files (for bad, ideally including an explanation of what's wrong, such as malformed header, truncated data segment, etc.) I think I could review this.

**Comment by neurolabusc** [comment]
Other than Matthew, I would think of @andrewjanke

**Comment by Kevin-Mattheus-Moerman** [comment]
@mfroeling @jhuguetn (Jordi can you also ask Paul Groot) want to review this submission for [JOSS](joss.theoj.org)? Or do you know anybody that could help? If you accept we'll open up a review issue. The review process follows our [review guidelines](http://joss.theoj.org/about#reviewer_guidelines). Let me know if you have any questions.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon assign @Kevin-Mattheus-Moerman as editor

**Comment by whedon** [comment]
OK, the editor is @Kevin-Mattheus-Moerman

**Comment by Kevin-Mattheus-Moerman** [comment]
@SebastianoF can you provide info in relation to @effigies 's question, i.e. do you have some tests/example Bruker files a reviewer could play with?

**Comment by SebastianoF** [comment]
@Kevin-Mattheus-Moerman There are bruker files a reviewer can play with, in the test_data folder.
They are a good starting point, and are the only one I could provide non-copyrighted at least at the moment. 
This may make the test self-referential. Having a non-copyrighted external (and so unbiased) dataset is not that straightforward. I heard offline that @neurolabusc had in mind to create a benchmarking platform with a wider dataset available, where the tests will be unbiased. He already pointed out a subtle issue of frame inversion that happens in ParaVision version 5.1, raised in the issue [#11](https://github.com/SebastianoF/bruker2nifti/issues/11).

**Comment by neurolabusc** [comment]
A nice repository of images is here
  https://gitlab.com/naveau/bruker2nifti_qa/tree/master
I think a challenge is that while the Bruker file format is thoroughly documented, the files created by Bruker software often omit required fields from some of the files. 
You are free to compare the results of this code versus my own
  https://github.com/neurolabusc/Bru2Nii
Since my code pre-dates this project (origins in 1990's) it tends to rely more on older Bruker files, and uses different heuristics to deal with missing fields. Of all my projects, this is the one I am least proud. Poorly conceived formats that are poorly executed necessarily lead to kludgey code. I think it is great that Sebastiano has created a clean-sheet implementation.

**Comment by rougier** [comment]
It's not exactly my expertise but I can (try to) help with review if needed.

**Comment by Kevin-Mattheus-Moerman** [comment]
@rougier Sure all help is welcome. I'm going to wait a couple of days to attract some more reviewers and then I'll open the formal review issue.

**Comment by andrewjanke** [comment]
I'm not entirely sure what is going on here, but I think I like what is being proposed. A code review for open journal publication of software?

I have a number of Bruker files here dating back to the early 90's, but as Chris points out the Bruker file format is rather like DICOM, it's a specification (effectively YAML but older) and how it has been used to determine orientation and other things from the acqup file is dependant on time, version, the author of the pulse sequence and a number of things. So 100% coverage is going to be hard. 

Also realise that orientation is not guaranteed if the pulse sequence author hasn't done things "right", for example the phase read direction might be reversed and acqp will not record this.

And then there is PV6...

Short version: Let me know if I can help although not sure of entirely what is being asked just yet.

**Comment by dzhoshkun** [comment]
Many thanks to all those who have replied so far. We would also very much appreciate any feedback and suggestions about the source code itself.

**Comment by oesteban** [comment]
Hi all, I think most of the people in this thread are better than me for this review, so I'll drop out. But thanks for the invite!

Best wishes to @sebastianof and this submission :)

**Comment by Kevin-Mattheus-Moerman** [comment]
@rougier do you mind if we open the official review issue with you as main reviewer? I'll continue to invite additional reviews though.

**Comment by rougier** [comment]
Ok. I can try to review but my review will be on a technical level mostly.

**Comment by Kevin-Mattheus-Moerman** [comment]
@rougier That is fine we can get started on the technical side first and I will recruit others to help with other aspects shortly

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon assign @rougier as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @rougier

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon start review

**Comment by whedon** [comment]
You didn't say the magic word! Try this:

```
@whedon start review magic-word=bananas
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon start review magic-word=bananas

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/354. Feel free to close this issue now!
